Index: test/testPositions.rb
===================================================================
--- test/testPositions.rb	(revision 1128)
+++ test/testPositions.rb	(working copy)
@@ -341,7 +341,7 @@
 list = [
 nil,
 nil, #['NewlineNode',0,1,0,6],
-  ['IterNode',0,0,4,6],
+  ['IterNode',0,0,0,6],
     ['FCallNode',0,0,0,6]
 ]
 test_tree(list, <<'END', "operation brace_block [paren-less no-args block")
@@ -351,7 +351,7 @@
 list = [
 nil,
 nil, #['NewlineNode',0,1,0,6],
-  ['IterNode',0,1,4,10],
+  ['IterNode',0,1,0,10],
     ['FCallNode',0,1,0,10]
 ]
 
@@ -363,7 +363,7 @@
 list = [
 nil,
 nil, #['NewlineNode',0,1,6,9],
-  ['IterNode',0,0,6,8],
+  ['IterNode',0,0,0,8],
     ['FCallNode',0,0,0,8],
       ['ArrayNode',0,0,3,5]
 ]
@@ -375,7 +375,7 @@
 list = [
 nil,
 nil, #['NewlineNode',0,1,6,9],
-  ['IterNode',0,1,6,12],
+  ['IterNode',0,1,0,12],
     ['FCallNode',0,1,0,12],
       ['ArrayNode',0,0,3,5]
 ]
@@ -388,7 +388,7 @@
 list = [
 nil,
 nil, #['NewlineNode',0,2,6,17],
-  ['IterNode',0,1,6,16],
+  ['IterNode',0,1,0,16],
     ['DAsgnNode',0,0,10,11],
     ['FCallNode',0,1,0,16],
       ['ArrayNode',0,0,3,5]
@@ -402,7 +402,7 @@
 list = [
 nil,
 nil, #['NewlineNode',0,2,6,19],
-  ['IterNode',0,1,6,18],
+  ['IterNode',0,1,0,18],
     ['MultipleAsgnNode',0,0,9,14],
       ['ArrayNode',0,0,10,13],
         ['DAsgnNode',0,0,10,11],
@@ -419,7 +419,7 @@
 list = [
 nil,
 nil, #['NewlineNode',0,2,10,23],
-  ['IterNode',0,1,10,22],
+  ['IterNode',0,1,0,22],
     ['MultipleAsgnNode',0,0,13,18],
       ['ArrayNode',0,0,14,17],
         ['DAsgnNode',0,0,14,15],
@@ -1282,7 +1282,7 @@
 list = [
 nil,
 nil,
-['IterNode', 0, 2, 15, 34],
+['IterNode', 0, 2, 0, 34],
 ['DAsgnNode', 0, 0, 19, 20],
 nil,
 ['FCallNode', 1, 1, 24, 30],
@@ -1460,7 +1460,7 @@
 list = [
 nil,
 nil, #['NewlineNode', 0, 1, 0, 25],
-['IterNode', 0, 0, 8, 24],
+['IterNode', 0, 0, 0, 24],
 nil, #['NewlineNode', 0, 0, 10, 24],
 ['FCallNode', 0, 0, 10, 22],
 ['ArrayNode', 0, 0, 15, 22],
Index: src/org/jruby/parser/DefaultRubyParser.java
===================================================================
--- src/org/jruby/parser/DefaultRubyParser.java	(revision 1128)
+++ src/org/jruby/parser/DefaultRubyParser.java	(working copy)
@@ -2583,7 +2583,7 @@
 case 351:
 					// line 1269 "DefaultRubyParser.y"
   {
-                  yyVal = new IterNode(support.union(((Token)yyVals[-4+yyTop]), ((Token)yyVals[0+yyTop])), ((Node)yyVals[-2+yyTop]), support.getCurrentScope(), ((Node)yyVals[-1+yyTop]), null);
+                  yyVal = new IterNode(support.union(((ISourcePositionHolder)yyVals[-5+yyTop]), ((Token)yyVals[0+yyTop])), ((Node)yyVals[-2+yyTop]), support.getCurrentScope(), ((Node)yyVals[-1+yyTop]), null);
                   support.popCurrentScope();
               }
   break;
@@ -2596,7 +2596,7 @@
 case 353:
 					// line 1275 "DefaultRubyParser.y"
   {
-                  yyVal = new IterNode(support.union(((Token)yyVals[-4+yyTop]), ((Token)yyVals[0+yyTop])), ((Node)yyVals[-2+yyTop]), support.getCurrentScope(), ((Node)yyVals[-1+yyTop]), null);
+                  yyVal = new IterNode(support.union(((ISourcePositionHolder)yyVals[-5+yyTop]), ((Token)yyVals[0+yyTop])), ((Node)yyVals[-2+yyTop]), support.getCurrentScope(), ((Node)yyVals[-1+yyTop]), null);
                   support.popCurrentScope();
               }
   break;
Index: src/org/jruby/parser/DefaultRubyParser.y
===================================================================
--- src/org/jruby/parser/DefaultRubyParser.y	(revision 1128)
+++ src/org/jruby/parser/DefaultRubyParser.y	(working copy)
@@ -1267,13 +1267,13 @@
 brace_block   : tLCURLY {
                   support.pushBlockScope();
 	      } opt_block_var compstmt tRCURLY {
-                  $$ = new IterNode(support.union($1, $5), $3, support.getCurrentScope(), $4, null);
+                  $$ = new IterNode(support.union($<ISourcePositionHolder>0, $5), $3, support.getCurrentScope(), $4, null);
                   support.popCurrentScope();
               }
               | kDO {
                   support.pushBlockScope();
 	      } opt_block_var compstmt kEND {
-                  $$ = new IterNode(support.union($1, $5), $3, support.getCurrentScope(), $4, null);
+                  $$ = new IterNode(support.union($<ISourcePositionHolder>0, $5), $3, support.getCurrentScope(), $4, null);
                   support.popCurrentScope();
               }
 
