Index: src/org/jruby/parser/ParserSupport.java
===================================================================
--- src/org/jruby/parser/ParserSupport.java	(revision 1135)
+++ src/org/jruby/parser/ParserSupport.java	(working copy)
@@ -238,7 +238,7 @@
         } else {
             switch (IdUtil.getVarType(id)) {
             case IdUtil.LOCAL_VAR:
-                return currentScope.assign(lhs.getPosition(), id, value);
+                return currentScope.assign(value != null ? union(lhs, value) : lhs.getPosition(), id, value);
             case IdUtil.CONSTANT:
                 if (isInDef() || isInSingle()) {
                     throw new SyntaxException(lhs.getPosition(), "dynamic constant assignment");
Index: src/org/jruby/ast/BlockArgNode.java
===================================================================
--- src/org/jruby/ast/BlockArgNode.java	(revision 1135)
+++ src/org/jruby/ast/BlockArgNode.java	(working copy)
@@ -83,7 +83,12 @@
         return name;
     }
 
+	public void setName(String name) {
+		this.name = name;
+	}
+	
     public List childNodes() {
         return EMPTY_LIST;
     }
+
 }
Index: src/org/jruby/ast/visitor/rewriter/ReWriteVisitor.java
===================================================================
--- src/org/jruby/ast/visitor/rewriter/ReWriteVisitor.java	(revision 1135)
+++ src/org/jruby/ast/visitor/rewriter/ReWriteVisitor.java	(working copy)
@@ -1681,6 +1681,7 @@
 	}
 
 	public Instruction visitRootNode(RootNode iVisited) {
+		config.getLocalVariables().addLocalVariable(iVisited.getStaticScope());
 		visitNode(iVisited.getBodyNode());
 		if(config.hasHereDocument()) {
 			config.fetchHereDocument().print();
