Index: src/org/jruby/ast/visitor/AbstractVisitor.java
===================================================================
--- src/org/jruby/ast/visitor/AbstractVisitor.java	(revision 1135)
+++ src/org/jruby/ast/visitor/AbstractVisitor.java	(working copy)
@@ -34,7 +34,9 @@
 import org.jruby.ast.AndNode;
 import org.jruby.ast.ArgsCatNode;
 import org.jruby.ast.ArgsNode;
+import org.jruby.ast.ArgsPushNode;
 import org.jruby.ast.ArrayNode;
+import org.jruby.ast.AttrAssignNode;
 import org.jruby.ast.BackRefNode;
 import org.jruby.ast.BeginNode;
 import org.jruby.ast.BignumNode;
@@ -130,7 +132,7 @@
  * @author  jpetersen
  */
 public abstract class AbstractVisitor implements NodeVisitor {
-    
+
     /**
      * This method is called by default for each visited Node.
      */
@@ -515,4 +517,12 @@
     public Instruction visitSymbolNode(SymbolNode iVisited) {
         return visitNode(iVisited);
     }
+    
+    public Instruction visitArgsPushNode(ArgsPushNode iVisited) {
+        return visitNode(iVisited);
+    }
+
+    public Instruction visitAttrAssignNode(AttrAssignNode iVisited) {
+        return visitNode(iVisited);
+    }
 }
Index: src/org/jruby/ast/visitor/NodeVisitor.java
===================================================================
--- src/org/jruby/ast/visitor/NodeVisitor.java	(revision 1135)
+++ src/org/jruby/ast/visitor/NodeVisitor.java	(working copy)
@@ -140,7 +140,7 @@
     public Instruction visitAndNode(AndNode iVisited);
     public Instruction visitArgsNode(ArgsNode iVisited);
     public Instruction visitArgsCatNode(ArgsCatNode iVisited);
-    public Instruction visitArgsPushNode(ArgsPushNode node);
+    public Instruction visitArgsPushNode(ArgsPushNode iVisited);
     public Instruction visitArrayNode(ArrayNode iVisited);
     public Instruction visitAttrAssignNode(AttrAssignNode iVisited);
     public Instruction visitBackRefNode(BackRefNode iVisited);
Index: src/org/jruby/ast/visitor/DefaultIteratorVisitor.java
===================================================================
--- src/org/jruby/ast/visitor/DefaultIteratorVisitor.java	(revision 1135)
+++ src/org/jruby/ast/visitor/DefaultIteratorVisitor.java	(working copy)
@@ -589,6 +589,7 @@
     
     public Instruction visitRootNode(RootNode iVisited) {
         iVisited.accept(_Payload);
+        iVisited.getBodyNode().accept(this);
         return null;
     }
 
